### ISSUE TITLE
2.0.4 support?

### ISSUE THREAD

**Issue by tommyettinger** [luapower/luajit#1]
LuaJIT 2.0.4 released (rather surprisingly) on May 14, 2015, after over a year of no releases. How difficult would it be to update LuaPower's version of LuaJIT to the latest?
I'm quite impressed by how easy it's been to get started with LuaPower, by the way. I'm planning to use the "Swedish Buffet" setup in my own repo, which if I'm not mistaken would be less than ideal to update the whole set of used packages frequently.  The mgit setup is definitely nice, but I would really like to be able to have some more control over my repo's layout (like having a README...), and I don't think I need to provide a module, more of an application.

**Comment by capr** [comment]
> LuaJIT 2.0.4 released (rather surprisingly) on May 14, 2015, after over a year of no releases. How difficult would it be to update LuaPower's version of LuaJIT to the latest?

luapower contains luajit 2.1 (newer than 2.0.4). I plan to upgrade it
to the latest HEAD soon, since an important (although very rare) bug
was fixed recently.

> I'm quite impressed by how easy it's been to get started with LuaPower, by the way.

Thanks, please keep me posted on whatever you find bad about it too :)
Feel free to use the forum too, I'm always looking to discuss/improve
things.

> I'm planning to use the "Swedish Buffet" setup in my own repo, which if I'm not mistaken would be less than ideal to update the whole set of used packages frequently.

Yes, you'll have to download them every time. A wget-based script to
automate that would be trivial to write, but nobody asked before :)
OTOH if we're talking about downloading a script to get started, then
I'm not sure why would you choose a wget-based script over mgit (which
is itself just a script) ?

> The mgit setup is definitely nice, but I would really like to be able to have some more control over my repo's layout (like having a README...)

I hear ya, the README thing is a bummer, but to have just that you'd
have to give up too much of the good stuff:
- using pandoc markdown instead of github's[1], so no more nice API tables
- loading modules and dependencies without a custom loader, so no
  more zero runtime overhead

[1] https://luapower.com/philosophy#documentation

**Comment by tommyettinger** [comment]
I wound up using a different copy of luapower-repos, modified to only pull from about a third of the repos, sticking the mgit scripts in my repo, and instructing users to run "mgit clone-all" to get the contents of those repos. I stuck all the things that the mgit command downloads in .gitignore so they don't wind up flooding my repo with old versions. This way things wind up where they should be, I only bundle the repos I need, I can have a README, etc. Things seem to be working great! Thanks again.

**Comment by capr** [comment]
> I wound up using a different copy of luapower-repos, modified to only pull from about a third of the repos, sticking the mgit scripts in my repo, and instructing users to run "mgit clone-all" to get the contents of those repos.

Cool, that's how it's meant to be used. Except maybe the scripts
deserve their own separate repo so that you can track their progress
too? I dunno.

Note that you can also make "releases" with mgit out of the current
state of your tree. A "release" is just a file containing a list of
packages[1] to be cloned by your users with `mgit clone-release
<release-name> | <release-file>`. I've seen them called "version
locks" in other places. Just a different way to skin the same cat.

> I stuck all the things that the mgit command downloads in .gitignore so they don't wind up flooding my repo with old versions. This way things wind up where they should be, I only bundle the repos I need, I can have a README, etc.

So basically your repo is normal git, and only the luapower repos are
overlaid, is that right? I mean you can have your README, but only for
your main repo, not for other repos? Not sure I got that last point (I
want README too, but I can't seem to find a nice way to have them
without complicating things badly).

> Things seem to be working great! Thanks again.

Glad to hear. And thanks for the feedback.

[1] https://github.com/luapower/forum/blob/master/.mgit/current.release
